### ISSUE TITLE
Multi project builds, run `resourceBasePath` different to test one

### ISSUE THREAD

**Issue by plippe** [scalatra/scalatra#834]
Hi,

I have a very simple [hello world app](https://github.com/plippe/scalatra-hello-resource). The main purpose of the app was to highlight how to read resource files. Those resources are stored in `src/main/webapp/WEB-INF`, and accessed with `getServletContext.getResource`. This worked perfectly until I tried with a multi project build.

`resourceBasePath` points to `PROJECT/src/main/webapp/WEB-INF` in run, but `src/main/webapp/WEB-INF` in test.

I was able to fix my tests with `servletContextHandler.setResourceBase(s"PROJECT/${resourceBasePath}")`, but I was wondering if this was the expected behaviour.

Kind regards

**Comment by takezoe** [comment]
JettyServer which is used in scalatra-test seems to need it in this case. I guess it's just a result, not intended behavior. So it might be possible to make it unnecessary in multi projects too by adding something to scalatra-test.

**Comment by plippe** [comment]
Thanks for the quick answer.

From what I am able to find, it all goes down to `Container.scala`:
- ScalatraFunSuite
- [ScalatraSuite](https://github.com/scalatra/scalatra/blob/3c7b7c3f24dbe334c0531a4241d09c97bb8da6b8/scalatest/src/main/scala/org/scalatra/test/scalatest/scalatest.scala#L61)
- [ScalatraTests](https://github.com/scalatra/scalatra/blob/3c7b7c3f24dbe334c0531a4241d09c97bb8da6b8/scalatest/src/main/scala/org/scalatra/test/scalatest/scalatest.scala#L13)
- [EmbeddedJettyContainer](https://github.com/scalatra/scalatra/blob/3c7b7c3f24dbe334c0531a4241d09c97bb8da6b8/test/src/main/scala/org/scalatra/test/ScalatraTests.scala#L12)
- [JettyContainer](https://github.com/scalatra/scalatra/blob/3c7b7c3f24dbe334c0531a4241d09c97bb8da6b8/test/src/main/scala/org/scalatra/test/EmbeddedJettyContainer.scala#L6)
- [Container](https://github.com/scalatra/scalatra/blob/4e4d4795dd152d4037b76e8e7bcb23c368b0f414/test/src/main/scala/org/scalatra/test/Container.scala)

Container hard codes [`resourceBasePath` to `src/main/webapp`](https://github.com/scalatra/scalatra/blob/4e4d4795dd152d4037b76e8e7bcb23c368b0f414/test/src/main/scala/org/scalatra/test/Container.scala#L7).

I haven't found a clean fix, but using the following works:
```
var resourceBasePath: String = {
  val projectPath = getClass.getClassLoader.getResource("").toURI.resolve("../../..")
   s"${projectPath}/src/main/webapp"
}
```

Would you like me to open a PR ?

**Comment by takezoe** [comment]
Does this work for non multi module projects?
Futhermore, I guess it doesn't work if the sub project is under a nested directory.

**Comment by plippe** [comment]
The fix ^^ seem to work for all types of projects.

https://github.com/plippe/scalatra-hello-resource = multi project
https://github.com/plippe/scalatra-hello-resource/pull/1 = single project root
https://github.com/plippe/scalatra-hello-resource/pull/2 = single project sub folder
https://github.com/plippe/scalatra-hello-resource/pull/3 = single project sub folders

`getClass.getClassLoader.getResource("")` returns the target folder for the project. `[ROOT OR PROJECT DIR]/target/scala-2.12/test-classes/`. 
`resolve("../../..")` goes up 3 folders into `[ROOT OR PROJECT DIR]`.

I can't seem to find a scenario where it wouldn't work, but I am happy to test any of your theories.

**Comment by takezoe** [comment]
Thanks for testing and pull request! I will test and merge it.

**Comment by plippe** [comment]
No problem, glad I could help.

**Comment by takezoe** [comment]
Fixed in #837
